.circleci/config.yml

```diff
@@ -75,106 +66,327 @@ commands:
+            fi
 
 jobs:
-  osteo:
+
+  conditionally-launch-build-and-store-all-job:
     executor:
       name: commit-executor
     steps:
-      - build-and-deploy-study:
+      - checkout
+      - conditionally-launch-build-and-store:
           study_key: osteo
-          study_guid: CMI-OSTEO
+      - conditionally-launch-build-and-store:
+          study_key: brain
+      - conditionally-launch-build-and-store:
+          study_key: angio
+      - conditionally-launch-build-and-store:
+          study_key: mbc
 
-  angio:
+  app-build-and-store-job:
     executor:
       name: commit-executor
     steps:
-      - build-and-deploy-study:
-          study_key: angio
-          study_guid: ANGIO
+      - app-build-and-store:
+          study_key: << pipeline.parameters.study_key >>
+
+  osteo-deploy-only:
+    executor:
+      name: commit-executor
+    steps:
+      - deploy-stored-build: &osteo-params
+          study_key: osteo
+          study_guid: CMI-OSTEO
 
-  brain:
+  brain-deploy-only:
     executor:
       name: commit-executor
     steps:
-      - build-and-deploy-study:
+      - deploy-stored-build: &brain-params
           study_key: brain
           study_guid: cmi-brain
 
-  mbc:
+  angio-deploy-only:
+    executor:
+      name: commit-executor
+    steps:
+      - deploy-stored-build: &angio-params
+          study_key: angio
+          study_guid: ANGIO
+
+  mbc-deploy-only:
     executor:
       name: commit-executor
     steps:
-      - build-and-deploy-study:
+      - deploy-stored-build: &mbc-params
           study_key: mbc
           study_guid: cmi-mbc
 
+  osteo-build-and-deploy:
+    executor:
+      name: commit-executor
+    steps:
+      - app-build-and-deploy:
+          <<: *osteo-params
+
+  angio-build-and-deploy:
+    executor:
+      name: commit-executor
+    steps:
+      - app-build-and-deploy:
+          <<: *angio-params
+
+  brain-build-and-deploy:
+    executor:
+      name: commit-executor
+    steps:
+      - app-build-and-deploy:
+          <<: *brain-params
+
+  mbc-build-and-deploy:
+    executor:
+      name: commit-executor
+    steps:
+      - app-build-and-deploy:
+          <<: *mbc-params
+
+parameters:
+  study_key:
+    type: string
+    default: "UNKNOWN"
+  do-builds:
+    type: boolean
+    default: false
+
 workflows:
   version: 2
-  build-deploy:
+
+  build-and-deploy-all-apps-workflow:
     jobs:
-      - osteo:
-          # todo: more DRY on branch filtering
-          filters:
-            branches:
-              only:
-                - develop
-      - angio:
+      - osteo-build-and-deploy: &build-and-deploy-filters
           filters:
             branches:
               only:
                 - develop
-      - brain:
+      - brain-build-and-deploy:
+          <<: *build-and-deploy-filters
+      - angio-build-and-deploy:
+          <<: *build-and-deploy-filters
+      - mbc-build-and-deploy:
+          <<: *build-and-deploy-filters
+
```

Comment:
Above is the "develop" workflow. If you trace back to each of the jobs (they run in parallel, by the way), you will find that I don't save tar archives for them. Seemed unnecessary, but welcome disagreements.